File: frontend/src/app/modules/bus/service/bus.service.ts
import {inject, Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BusModel} from '../model/bus.model';
import {JsonConvert} from 'json2typescript';
import {PageResponse} from '../../../shared/page/page-response';

@Injectable({providedIn: 'root'})
export class BusService {

  private http = inject(HttpClient);
  private converter = new JsonConvert();

  getListBuses(pageFirst: number, pageLast:number){
    return this.http.get<PageResponse<BusModel>>('/bus/load', {params: {first: pageFirst, last: pageLast}});
      // .pipe(map(data => this.converter.deserialize(data, BusModel)));
  }

  getBusById(busId: number){
    return this.http.get<BusModel>(`/bus/${busId}`);
  }

  createBus(bus: BusModel){
    return this.http.post<BusModel>(`/bus`, this.converter.serialize(bus, BusModel));
  }

  updateBus(busId: number, bus: BusModel){
    return this.http.put<BusModel>(`/bus/${busId}`, this.converter.serialize(bus, BusModel));
  }

  deleteBus(busId: number){
    return this.http.delete<BusModel>(`/bus/${busId}`);
  }

}

File: src/main/java/staj/spring/ticket_webapp/bus/BusService.java
package staj.spring.ticket_webapp.bus;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface BusService {

    Page<BusDTO> getAllBuses(Pageable pageable);

    Page<BusDTO> getAllBuses(Integer first, Integer last);

    BusDTO getBusById(Long id);

    BusDTO createBus(BusDTO busDTO);

    BusDTO updateBus(Long id, BusDTO busDTO);

    void deleteBus(Long id);

    BusDTO getBusByPlate(String plate);
}
